apps/web/app/components/providers/account-watcher-provider.tsx

```diff
@@ -0,0 +1,21 @@
+import { type ReactNode, useEffect } from "react";
+import { useConfig } from "wagmi";
+import { watchAccount } from "wagmi/actions";
+import useUserRoleCookie from "../hooks/use-user-role-cookie";
+
+export default function AccountWatcherProvider({ children }: { children: ReactNode }) {
+  const config = useConfig();
+  const { cookie, setCookie } = useUserRoleCookie();
+
+  useEffect(() => {
+    return watchAccount(config, {
+      onChange(account, prevAccount) {
+        if (account.address !== prevAccount.address && account.address !== undefined) {
```

Comment:
why we don't do anything if the acc.address is undefined? Shouldn't the cookie be clear?

Reply:
We can do that, but it shouldn't affect app behaviour.

Reply:
If we don't, I would love a comment explaining why it's not needed. I feel that it's not obvious.